### PR TITLE
Add initial workspace creator as workspace enterprise app owner

### LINKED ISSUE #2625: With auto_create no interactive user is the owner of the Workspace Enterprise App
When creating a workspace with `auto_create` the workspace owner is set to the `Application Admin` account. This means that to add any users to the App Roles the interactive user must have privileges to access all Enterprise Apps in Azure AD, or ask their AD Admin to grant them access.

I suggest the creator of the workspace is also made an owner of the Workspace Enterprise application. This will mean they can also view sign in logs.

The user needs adding here: https://github.com/marrobi/AzureTRE/blob/caf6a9c2e68f7e74684249d112e2b083f77ac3a9/templates/workspaces/base/terraform/aad/aad.tf#L97

### PR DESCRIPTION
**Pull request by marrobi**
# Resolves #2625

## How is this addressed

- Add initial workspace creator as owner of the enterprise app
- Give Application Admin `Directory.Read.All` permissions in `make auth`

## Breaking change.
The Application Admin account needs to be granted `Directory.Read.All` permissions:

image

### CODE REVIEW

**Comment by marrobi** [comment]
It seems this need an additional permissions to be able to add this user. It worked in my environment, but  failed for a customer. I have the some additional permissions -  `Directory.Read.All` and `Group.ReadWrite.All`. I am guessing this operation tries to read the workspace owner object and hence needs `Directory.Read.All`

**Comment by marrobi** [comment]
`Directory.Read.All` fixes the deployment, although user still cannot access the enterprise app n the portal

**Comment by marrobi** [comment]
User also needs to have "Directory Reader" role in AD.

File: docs/tre-admins/identities/application_admin.md [hunk 1]
```diff
@@ -10,8 +10,8 @@ This application does not have any roles defined.
...
-| Application.ReadWrite.All | Application | Yes | If the AAD Administrator has delegated AAD administrative operations to the TRE, then this user should be granted `Application.ReadWrite.All`. This will allow the user to create workspace applications and administer any applications in the tenant. There will be no need for the Tenant Admin to oversee the Tenant. |
-| Directory.Read.All | Application | Yes | This permission is required to read User details from Azure Active Directory. This is requried if Azure AD groups are to be created automatically by the TRE. |
+| Application.ReadWrite.All | Application | Yes | This permission is required to create workspace applications and administer any applications in the tenant. IThis is needed if the AAD Administrator has delegated AAD administrative operations to the TRE. There will be no need for the Tenant Admin to manually create workspace applications in the Tenant. |
```
Comment (martinpeck) [inline docs/tre-admins/identities/application_admin.md:13]: Typo `IThis` in second sentence.

**Review by martinpeck — APPROVED** [review]
Approved, but you have a typo

**Comment by marrobi** [comment]
/test-force-approve passed https://github.com/microsoft/AzureTRE/actions/runs/3150965693

**Pull request merged**